helm/templates/deployment.yaml

```diff
@@ -66,18 +66,73 @@ spec:
           mountPath: /servo/servo.yaml
           subPath: servo.yaml
           readOnly: true
+        {{- if eq .Values.connectionSettingsType "sidecar" }}
+        - name: agent-management-sidecar
+          mountPath: /agent-management-sidecar
+        {{- end }}
         resources:
           limits:
             cpu: 250m
             memory: 512Mi
+      {{- if eq .Values.connectionSettingsType "sidecar" }}
+      - name: agent-management-sidecar
+        image: {{ required ".Values.agentMgmtSidecarImage is required for sidecar connection settings type" .Values.agentMgmtSidecarImage }}
+        resources:
+          limits:
+            cpu: 100m
+            memory: 128Mi
+        env:
+        - name: SERVICE_URL
+          value: {{ default "<SERVICE_URL>" .Values.agentMgmtServiceUrlOverride }}
```

Comment:
Is this an acceptable default? (same for other uses of default "<...>")

Reply:
Separate note on naming - I see variables named `somethingOverride`, can we rename them to just `something` and keep them optional.

Reply:
>Is this an acceptable default? (same for other uses of default "<...>")

This is the format expected by the orchestrator in order to perform substitution. Here's the slack thread for reference https://appdynamics.slack.com/archives/C0468BYM85C/p1675881169951929?thread_ts=1675877822.302409&cid=C0468BYM85C

>I see variables named somethingOverride

This makes it clear the end user is not expected to provide these values since the whole point of this configuration is to have them populated externally.